src/models/MediaMetadata.ts

```diff
@@ -1,62 +1,79 @@
 import * as mongoose from 'mongoose';
 import { Schema, Document } from 'mongoose';
+import { ValidationError } from '../helpers/customErrors';
 
 const DOCUMENT_EXPIRY_IN_SECONDS = 2592000; // 30 days
 
 mongoose.set('useCreateIndex', true);
 
 export interface MediaMetadataInterface extends Document {
-  subcount?: string;
-  title?: string;
-  director?: string;
-  imdbID?: string;
-  osdbHash: string;
-  genres?: Array<string>;
-  actors?: Array<string>;
-  episodeTitle?: string;
-  seasonNumber?: string;
+  actors: Array<string>;
+  directors: Array<string>;
   episodeNumber?: string;
-  year?: string;
-  type?: string;
+  episodeTitle?: string;
+  genres: Array<string>;
   goofs?: string;
-  trivia?: string;
+  imdbID: string;
+  osdbHash: string;
+  seasonNumber?: string;
   tagline?: string;
+  title: string;
+  trivia?: string;
+  type: string;
+  year: string;
 
   // Added automatically:
   createdAt: string;
   updatedAt: string;
 }
 
+const isRequiredFieldForTVEpisodesIncluded = (fieldToValidate: string): boolean => {
```

Comment:
Keen to see some unit test coverage for this, mainly to ensure the use of arrow functions is okay - are we sure `this` is the document?